crypto/evp/mac_meth.c

```diff
@@ -193,6 +205,28 @@ const OSSL_PARAM *EVP_MAC_settable_ctx_params(const EVP_MAC *mac)
     return mac->settable_ctx_params(ossl_provider_ctx(EVP_MAC_provider(mac)));
 }
 
+const OSSL_PARAM *EVP_MAC_CTX_gettable_params(EVP_MAC_CTX *ctx)
+{
+    void *prov_ctx = ossl_provider_ctx(EVP_MAC_provider(ctx->meth));
+
+    if (ctx->meth->ctx_gettable_params != NULL)
+        return ctx->meth->ctx_gettable_params(prov_ctx, ctx->data);
```

Comment:
why do this if we dont resolve it?

Reply:
??? I don't understand

Reply:
The resolve does this
````
        case OSSL_FUNC_MAC_CTX_GETTABLE_PARAMS:
            if (mac->ctx_gettable_params != NULL)
                break;
````

So why check later if it is NULL and call it when it is always going to be NULL.

Reply:
The resolve is checking for a double definition.  If it isn't defined at all it can still be NULL, so there needs to be a check here.